charts/k8ssandra/templates/crd/batch_job.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-crd-upgrader-job-k8ssandra
+  labels: {{ include "k8ssandra.labels" . | indent 4 }}
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded
```

Comment:
It might be good to make the deletion policy configurable and consider making the default `before-hook-creation`  so that if things do go sideways the user will have logs.